x-pack/plugins/maps/public/components/map/mb/view.js

```diff
@@ -32,55 +41,76 @@ export class MBMapContainer extends React.Component {
     }
   }, 256);
 
-  _updateTooltipState = _.debounce(async (e) => {
 
-    const mbLayerIds = this._getMbLayerIdsForTooltips();
-    const features = this._mbMap.queryRenderedFeatures(e.point, { layers: mbLayerIds });
+  _lockTooltip = async (e) => {
+
+    const features = this._getFeaturesUnderPointer(e.point);
+    if (!features.length) {
+      this.props.setTooltipState(null);
+      return;
+    }
+
+    const targetFeature = features[0];
+    const layer = this._getLayer(targetFeature.layer.id);
+    const popupAnchorLocation = this._justifyAnchorLocation(e.lngLat, targetFeature);
+    this.props.setTooltipState({
```

Comment:
Should this function also call `this. _updateHoverTooltipState.cancel();` to cancel any debounce update hover tooltip state events. There could be a race condition where mouse move event is debouncing, click event is fired and state is updated, and then mouse move debounce fires prior to updated state getting pushed to component

Reply:
yes! need to fix, thx